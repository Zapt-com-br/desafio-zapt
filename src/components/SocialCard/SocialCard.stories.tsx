import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import SocialCard, { SocialCardProps } from '.';

export default {
  title: 'Social Comments/SocialCard',
  component: SocialCard,
} as Meta;

export const SocialCardDefault: Story<SocialCardProps> = args => (
  <SocialCard {...args} />
);

SocialCardDefault.args = {
  username: '@LuisCarlos',
  colorUserProfile: 'black',
  imageProfile:
    'https://www.pixsy.com/wp-content/uploads/2021/04/ben-sweet-2LowviVHZ-E-unsplash-1.jpeg',
  comment:
    '”Não da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um só dia... O pessoal aqui de casa ficou viciado de um nível que terei que comprar toda semana... Assim não tem academia que secura rsrsrsrs Obrigado Zapt.”',
};
