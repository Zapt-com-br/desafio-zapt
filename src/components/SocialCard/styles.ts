import styled, { css } from 'styled-components';
import { SocialCardProps } from '.';

type WrapperProps = Pick<SocialCardProps, 'transparent' | 'maxSizeCard'>;

const isTransparent = (condition: boolean, color: string) =>
  condition ? 'transparent' : color;

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, transparent, maxSizeCard }) => css`
    max-width: ${maxSizeCard ? '46rem' : '38rem'};
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};
    background: ${isTransparent(transparent!, theme.colors.gradients.lightBg)};
    break-inside: avoid;
  `}
`;
export const UserInformation = styled.div``;
export const Comment = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;
