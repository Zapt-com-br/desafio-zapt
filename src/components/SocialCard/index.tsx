import React from 'react';

import { Typography, UserProfile } from 'components';

import * as Styled from './styles';

type Sizes = 'small' | 'medium' | 'xmedium' | 'large';

export type SocialCardProps = {
  comment: string;
  username: string;
  imageProfile: string;
  userSubtext?: string;
  transparent?: boolean;
  textColor?: 'black' | 'darkgray' | 'white' | 'primary';
  colorUserProfile?: 'black' | 'darkgray' | 'white' | 'primary';
  commentSize?: Sizes;
  usernameSize?: Sizes;
  maxSizeCard?: boolean;
};

const SocialCard = ({
  imageProfile,
  username,
  comment,
  transparent = false,
  maxSizeCard = false,
  textColor = 'primary',
  colorUserProfile = 'primary',
  userSubtext = '',
  commentSize = 'medium',
  usernameSize = 'medium',
}: SocialCardProps) => (
  <Styled.Wrapper maxSizeCard={maxSizeCard} transparent={transparent}>
    <Styled.Comment>
      <Typography size={commentSize} color={textColor}>
        {comment}
      </Typography>
    </Styled.Comment>
    <Styled.UserInformation>
      <UserProfile
        usernameSize={usernameSize}
        subtext={userSubtext}
        textColor={colorUserProfile}
        username={username}
        imageProfile={imageProfile}
      />
    </Styled.UserInformation>
  </Styled.Wrapper>
);

export default SocialCard;
