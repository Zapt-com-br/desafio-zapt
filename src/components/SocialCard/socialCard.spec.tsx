import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'helpers/renderWithThame';

import SocialCard, { SocialCardProps } from '.';

jest.mock('components/UserProfile', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="userProfile"></div>;
    },
  };
});

const props: SocialCardProps = {
  comment: 'test comment',
  username: 'test name',
  imageProfile: 'image ',
};

describe('<SocialCard />', () => {
  it('should render SocialCard', () => {
    const { container } = renderWithTheme(<SocialCard {...props} />);

    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByTestId('userProfile')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render SocialCard with size default', () => {
    const { container } = renderWithTheme(<SocialCard {...props} />);

    expect(container.firstChild).toHaveStyleRule('max-width', '38rem');
  });

  it('should render SocialCard with full size if passed', () => {
    const { container } = renderWithTheme(
      <SocialCard maxSizeCard {...props} />
    );

    expect(container.firstChild).toHaveStyleRule('max-width', '46rem');
  });

  it('should render SocialCard with backgroung', () => {
    const { container } = renderWithTheme(<SocialCard {...props} />);

    expect(container.firstChild).toHaveStyle({
      background:
        'linear-gradient(0deg, rgba(255, 255, 255, 0.9), rgba(255, 255, 255, 0.9)), #2196F3',
    });
  });

  it('should render SocialCard without backgroung', () => {
    const { container } = renderWithTheme(
      <SocialCard transparent {...props} />
    );

    expect(container.firstChild).toHaveStyle({
      background: 'transparent',
    });
  });
});
