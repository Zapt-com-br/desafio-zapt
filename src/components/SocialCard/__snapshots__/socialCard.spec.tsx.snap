// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialCard /> should render SocialCard 1`] = `
.c2 {
  font-weight: 400;
  color: #011FBB;
  font-size: 1.6rem;
}

.c0 {
  max-width: 38rem;
  padding: 1.6rem 2.4rem;
  background: linear-gradient(0deg,rgba(255,255,255,0.9),rgba(255,255,255,0.9)),#2196F3;
  -webkit-break-inside: avoid;
  break-inside: avoid;
}

.c1 {
  margin-bottom: 1.6rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="c2"
      color="primary"
    >
      test comment
    </span>
  </div>
  <div
    class=""
  >
    <div
      data-testid="userProfile"
    />
  </div>
</div>
`;
